ecommerce/core/management/commands/generate_courses.py

```
@@ -0,0 +1,139 @@
+                continue
+            try:
+                for seat in course_settings["seats"]:
+                    self._create_seat(course, seat, partner)
+            except KeyError:
+                self._abort_course_generation(course, "Course JSON object is missing required seat fields")
+                continue
+
+            # Publish the data to the LMS
+            course.publish_to_lms()
+
+    def _abort_course_generation(self, course, error):
+        """ Deletes a course having provisioning problems """
+        logger.warning("Can't create course, proceeding to next course")
+        if course is not None:
+            course.delete()
+            msg = "Deleted course with id %s" % (course.id)
+            logger.info(msg)
+        logger.warning(error)
+
+    def _create_seat(self, course, seat, partner):
+        """ Add the specified seat to the course """
+        seat_type = seat["seat_type"]
+        if seat_type == "audit":
+            course.create_or_update_seat("", False, 0, partner)
+        elif seat_type == "verified":
+            # TODO - Enable verification deadline if developers want it
+            course.create_or_update_seat(
+                "verified",
+                True,
+                self.default_verified_price,
+                partner,
+                expires=self.default_upgrade_deadline
+            )
+        elif seat_type == "honor":
+            course.create_or_update_seat("honor", False, 0, partner)
+        elif seat_type == "professional":
+            # TODO - Enable verification deadline if developers want it
+            course.create_or_update_seat(
+                "professional",
+                seat["id_verification_required"],
+                self.default_professional_price,
+                partner
+            )
+        elif seat_type == "credit":
+            # Requires manually creating TestCreditProvider in the LMS django admin
```

Should this raise an Exception if TestCreditProvider is not set?